code/modules/library/computers/base.dm

```diff
@@ -0,0 +1,93 @@
+#define MAX_BOOK_FLAGS	3	// maximum number of times a book can be flagged before being removed from results
+
+/obj/machinery/computer/library
+	name = "visitor computer"
+	anchored = 1
+	density = 1
+	icon_keyboard = ""
+	icon_screen = "computer_on"
+	var/screenstate = 0
+	var/page_num = 1
+	var/num_pages = 0
+	var/num_results = 0
+	var/datum/library_query/query = new()
+
+	icon = 'icons/obj/library.dmi'
+	icon_state = "computer"
+
+/obj/machinery/computer/library/proc/interact_check(var/mob/user)
+	if(stat & (BROKEN | NOPOWER))
+		return 1
+
+	if (!Adjacent(user))
+		if (!issilicon(user) && !isobserver(user))
+			user.unset_machine()
+			user << browse(null, "window=library")
+			return 1
+
+	user.set_machine(src)
+	return 0
+
+/obj/machinery/computer/library/proc/get_page(var/page_num)
+	var/searchquery = ""
+	var/where = 0
+	if(query)
+		if(query.title && query.title != "")
+			searchquery += " WHERE title LIKE '%[sanitizeSQL(query.title)]%'"
```

Comment:
@Crazylemon64 I added the SQL sanitizations to the actual query code here so it should be bulletproof; I'm pretty sure running `sanitizeSQL()` twice on text with a & in it will turn it into &amp;amp; or something, so I'll leave the earlier settings of those vars be.